packages/ds-ext/src/app/VsCodeClient.tsx

```diff
@@ -5,11 +5,13 @@ import type { WorkspaceApiClient, ClientRunParams, ServerClientObservationConfig
 export class VsCodeClient implements WorkspaceApiClient {
   updateEdgeCounts: any;
   private vscode: any;
-  private observers: ServerClientObservationConfig | undefined;
+  private observers: ServerClientObservationConfig|undefined;
+
   constructor(vscode: any) {
     this.vscode = vscode;
 
     window.addEventListener('message', (event) => this.handleMessage(event.data));
+    this.run = this.run.bind(this);
```

Comment:
Doesent this line imply we are still de facto using detached methods from a class? Even though binding `this` will allow it to keep acting as a class method, its feels like a slight hack?

Reply:
Yes, in regular method functions (non-arrow functions), the value of `this` is determined at the time the function is "called," rather than at the time of its "definition." Therefore, the calling context influences the value of `this`. In `this.run.bind(this);`, we specify that when calling `run`, it should use the `this` from `bind(this)`.